docs/Developer-guide.md

```diff
@@ -0,0 +1,131 @@
+# Developer guide
+
+The following document details the proposed structure for this repo, the benchmark scripts and libraries.
+
+## General design and motivations
+
+The goal of the benchmark repo is to provide scripts to compare tree-based machine learning framework on similar tasks and environment.
+
+The benchmark is made of a collection of scripts. These scripts can be run locally, or manually from within a remote VM. But these scripts will eventually be submitted as a pipeline or workflow chaining scripts in a sequence: data generation, data processing, training, inferencing.
+
+In order to do that, we need each step for benchmarking a given framework to be implemented as a single script with clear inputs, outputs and parameters. We organize those into sub-directories under `/src/scripts/`.
+
+For the benchmark to automate the analysis of the results and the production of a benchmark report, we need each script to report a common set of metrics or tags.
+
+## Proposed structure of the repo
+
+| Directory | Content |
+| :-- | :-- |
+| `/.github/workflows/` | [github workflows](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions) for this project |
+| `/docs/` | Documentation in markdown |
+| `/pipelines/` | (coming) Scripts or config files for running the benchmark in the cloud (ex: AzureML) |
+| `/src/` | &nbsp; |
+| `/src/common/` | Common libraries for all benchmark scripts |
+| `/src/scripts/` | Benchmark scripts |
+| `/tests/` | Unit tests using pytest |
+
+## Proposed structure of the scripts
+
+**Scripts will be written in python**. Even if the module is calling a CLI command (ex: lightgbm.exe), writing it in python will allow us to wrap it with metrics reporting, consistent unit tests, etc.
```

Comment:
Should each script specify it's own Conda / Python environment?

Reply:
For running the benchmark as part of an orchestrated pipeline, definitely. But for manual/local use, having one environment for all of them would be better.

Maybe that's where the main `requirements.txt` file at the root can be used for manual/local, while each script will provide a conda env in yaml.

Reply:
I can see a tension here between local/manual use of the benchmark scripts, in which users will create one environment for all of them (like we do for unit testing), and the definition of distinct conda environments for each script.